From 7c4c94e83bc8e2f8922a1bd2cebb54a08d1724a2 Mon Sep 17 00:00:00 2001
From: Alejandro Lozano <alejandro.lozano@freescale.com>
Date: Wed, 27 Apr 2016 18:55:52 -0500
Subject: [PATCH 15/23] Add Murata Wifi ZP support for the EVB-SCMIMX6SX

Connector Rev2. WL_REG_ON is connected via the Bluetooth
J15 connector. R312 from SCH-29199 REV A must be populated.

Signed-off-by: Alejandro Lozano <alejandro.lozano@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sdb.dts                   |  2 +-
 arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts   | 36 ++++++++++++++++++++++
 arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts | 34 ++++++++++++++++++++
 3 files changed, 71 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts

diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 1c8992e..73694e7 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -467,9 +467,9 @@
 &usdhc2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc2>;
-	non-removable;
 	no-1-8-v;
 	keep-power-in-suspend;
+	wifi-host;
 	enable-sdio-wakeup;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
new file mode 100644
index 0000000..1658385
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-1gb-evb-ldo.dts"
+
+
+/ {
+	regulators {
+		wlreg_on: fixedregulator@100 {
+			compatible = "regulator-fixed";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-name = "wlreg_on";
+			gpio = <&gpio4 8 0>;
+			startup-delay-us = <100>;
+			enable-active-high;
+		};
+	};
+
+	bcmdhd_wlan_0: bcmdhd_wlan@0 {
+		compatible = "android,bcmdhd_wlan";
+		wlreg_on-supply = <&wlreg_on>;
+	};
+};
+
+
+&usdhc2 {
+	pinctrl-0 = <&pinctrl_usdhc2>;
+};
+
+
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts
new file mode 100644
index 0000000..291e37f
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-512mb-evb-ldo.dts"
+
+/ {
+	regulators {
+		wlreg_on: fixedregulator@100 {
+			compatible = "regulator-fixed";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-name = "wlreg_on";
+			gpio = <&gpio4 8 0>;
+			startup-delay-us = <100>;
+			enable-active-high;
+		};
+	};
+
+	bcmdhd_wlan_0: bcmdhd_wlan@0 {
+		compatible = "android,bcmdhd_wlan";
+		wlreg_on-supply = <&wlreg_on>;
+	};
+};
+
+
+&usdhc2 {
+	pinctrl-0 = <&pinctrl_usdhc2>;
+};
+
-- 
1.9.1

